src/steps/scroll-through.ts

```diff
@@ -0,0 +1,33 @@
+import { BaseStep, Field, StepInterface } from '../core/base-step';
+import { Step, RunStepResponse, FieldDefinition, StepDefinition } from '../proto/cog_pb';
+
+export class ScrollThrough extends BaseStep implements StepInterface {
+
+  protected stepName: string = 'Scroll through a percentage of a web page';
+  // tslint:disable-next-line:max-line-length
+  protected stepExpression: string = 'scroll through (?<depth>\\d+)% of the page';
+  protected stepType: StepDefinition.Type = StepDefinition.Type.ACTION;
+  protected expectedFields: Field[] = [{
+    field: 'depth',
+    type: FieldDefinition.Type.NUMERIC,
+    description: 'Percent Depth',
+  }];
+
+  async executeStep(step: Step): Promise<RunStepResponse> {
+    const stepData: any = step.getData().toJavaScript();
+    const depth: number = stepData.depth;
+
+    try {
+      await this.client.scrollThrough(depth);
+      return this.pass('Successfully scrolled through: %s percent depth', [depth]);
```

Comment:
Might recommend language mirroring the step expression here (and in the error response below): `Successfully scrolled through %d%% of the page`

Also note: `%%` becomes `%` (and `%d` can be used for float/int values).